### ISSUE TITLE
Metadata File Not Parsed

### ISSUE THREAD

**Issue by isosphere** [noembryo/KoHighlights#29]
The attached file exists in a folder structure that contains many other folders with similar files, all of which are processed by KOHighlights and rendered in the GUI. This one alone is not parsed at all - it does not appear in the GUI.

[metadata.epub.lua.zip](https://github.com/noembryo/KoHighlights/files/13228064/metadata.epub.lua.zip)

I've tried removing the multi-line strings on a lark, but that had no effect. The folder name is `Lord Foul's Bane - Stephen R. Donaldson (1646).sdr` - I've tried removing the apostrophe, to no avail. 

It is not archived. 

When I use the slppu dependency directly to decode this file, the result is simply `return`. 

```python
from slppu import slppu as lua

data = None
print("Reading file...")

with open(r'C:\Users\Matt\Downloads\test\metadata.epub.lua', encoding='utf8') as file:
    data = file.read()

print(f"Data read, it's {len(data)} characters long.")

assert isinstance(data, str), "data isn't a string"

print("Parsing file...")
data = lua.decode(data)

print("Parsed:")
print(data)
```

> Reading file...
> Data read, it's 4549 characters long.
> Parsing file...
> Parsed:        
> return

Though I get the same result for any file - I can't seem to get SLPPU to parse anything on its own. The upstream repo uses something called `loadstring` which is not included and I am not familiar with being part of the Python standard library, maybe that's important?

**Comment by noembryo** [comment]
I'll have a look, but I was wondering if you had made any recent updates to KOReader and first open the book with it, because there where some planed changes to the way the Highlights are stored in the metadata files.
What is your current version?

**Comment by isosphere** [comment]
> What is your current version?

I'm on the latest release, 2023.10, released a couple days ago. That might explain it!

**Comment by noembryo** [comment]
Did you actually opened this book *after* the update?

**Comment by isosphere** [comment]
As a test, I've opened a book that previously showed up in the KoHighlights GUI, highlighted something randomly, and tried to parse it again. That book disappeared. I think you're right about this being a format change issue with this release.

**Comment by noembryo** [comment]
I was afraid of this..
Well, it's time to try to remember this code so I can update it..
It will be very helpful to me if you could send me any samples that you have of the new metadata, until I install the newer version myself..

**Comment by isosphere** [comment]
Here is a zipped archive of the same metadata file before and after its modification by the new version, which I hope you find helpful. Unfortunately this is about all I have to share at the moment - this is a new workflow for me, and I've only just started using KOReader.

[metadata.pdf.lua.zip](https://github.com/noembryo/KoHighlights/files/13228835/metadata.pdf.lua.zip)

**Comment by noembryo** [comment]
OK, I think its enough for starters.
When I find some time I will install it in my reader too.
Thank you for your help.. 😃

**Comment by noembryo** [comment]
Fixed in [v1.7.3.0](https://github.com/noembryo/KoHighlights/releases/tag/v1.7.3.0).
